pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -395,6 +392,35 @@ func (p *provider) get(machine *v1alpha1.Machine) (*doInstance, error) {
 	return nil, cloudprovidererrors.ErrInstanceNotFound
 }
 
+func (p *provider) listDroplets(token string) ([]godo.Droplet, error) {
+	ctx := context.TODO()
+	client := getClient(token)
+	result := make([]godo.Droplet, 0)
+	page := 1
```

Comment:
https://github.com/digitalocean/godo
has an slightly simpler example for this paging. it does not start counting at 1 but 0.

Reply:
Yeah I'll make it prettier, I was in a haste when I pushed this.

Reply:
It's now a hypertext enabled HATEOAS client.